unit_tests/stencil_composition/structured_grids/test_tmp_storage_sid_cuda.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+ * GridTools
+ *
+ * Copyright (c) 2014-2019, ETH Zurich
+ * All rights reserved.
+ *
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include "../test_tmp_storage_sid_cuda.hpp"
+
+#include <gridtools/stencil_composition/backend_cuda/tmp_storage_sid.hpp>
+#include <gridtools/stencil_composition/sid/concept.hpp>
+#include <gridtools/tools/backend_select.hpp>
+
+#include <type_traits>
+
+#include "../../tools/multiplet.hpp"
+
+#include <gtest/gtest.h>
+
+namespace gridtools {
+    namespace {
+        constexpr int_t extent_i_minus = -1;
+        constexpr int_t extent_i_plus = 2;
+        constexpr int_t extent_j_minus = -3;
+        constexpr int_t extent_j_plus = 4;
+
+        constexpr int_t blocksize_i = 32;
+        constexpr int_t blocksize_j = 8;
+
+        template <typename T, typename Allocator> // test fixture works with default constructible allocators
+        class tmp_cuda_storage_sid : public ::testing::Test {
```

Comment:
I don't see the point of writing test fixture here.
You just distribute input parameters for your the only one testing scenario over three different places -- global staff; members of the fixture; local variables in the test function.

Reply:
Yes, the test fixture is an artifact of the previous test structure...